### PR TITLE
[IOU] Pay with Expensify option

### PR DESCRIPTION
**Pull request by jasperhuangg**
### Details
Added the option to pay with Expensify in the IOU Details modal if the report has USD set as its currency. See the issue for more context.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/159572

### Tests
1. Created 2 users (A, and B) with Gold Expensify Wallets (via [this SO on dev](https://stackoverflow.com/c/expensify/questions/10083) or [this SO on prod](https://stackoverflow.com/c/expensify/questions/9868/9869#9869)). Also, make sure they have deposit bank accounts set up by running `./script/clitools.sh generator:bankaccount`.
2. Used A to submit an IOU Request to B.
3. Used the Pay with Expensify option to pay for B.
4. Verified that the payments were added in the reimbursements table:
```SQL
SELECT * from reimbursements WHERE reportID=[reportID];
```  
5. Verify that the UI is updated to reflect that the IOU report was settled (see screenshots). There should be a report action "Settled up using Expensify".

### QA Steps
Same as the above tests.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web
Screen Shot 2021-05-27 at 9 57 13 AM
Screen Shot 2021-06-02 at 11 35 38 AM

#### Mobile Web
Simulator Screen Shot - iPhone 11 - 2021-05-27 at 10 04 49
Simulator Screen Shot - iPhone 11 - 2021-05-27 at 10 04 55
Screen Shot 2021-06-02 at 11 40 06 AM

#### Desktop
Screen Shot 2021-05-27 at 9 59 10 AM
Screen Shot 2021-06-02 at 11 35 38 AM

#### iOS
Simulator Screen Shot - iPhone 11 - 2021-05-27 at 10 02 43
Simulator Screen Shot - iPhone 11 - 2021-05-27 at 10 02 46
Screen Shot 2021-06-02 at 11 45 24 AM

#### Android
171622081774_ pic
181622081775_ pic
Screen Shot 2021-06-02 at 11 46 14 AM

### CODE REVIEW

File: src/pages/iou/IOUDetailsModal.js [hunk 1]
```diff
@@ -173,10 +175,28 @@ class IOUDetailsModal extends Component {
             });
     }
 
+    /**
+     * Checks to see if we can use Expensify.
```
Comment (Julesssss) [inline src/pages/iou/IOUDetailsModal.js:179]: > Checks to see if we can use Expensify.

Could you expand on this a bit? 'use Expensify' without the pay-with-expensify context is a bit vague IMO.

**Comment by jasperhuangg** [comment]
Hey @Julesssss! My bad! Forgot to update the testing instructions. You also need to run `./script/clitools.sh generator:bankaccount` to generate a deposit bank account for the payer.

It's because the PayWithWallet command was recently updated to [require either a default bank account or debit card fund for the payer](https://github.com/Expensify/Auth/blob/e93624679ec767f0014ec8df5bdb25a8f3356851/auth/command/PayWithWallet.cpp#L58-L66). I think it's so we can support Bancorp personal payments?

**Review by Julesssss — CHANGES_REQUESTED** [review]
Thanks, I was able to settle the IOU!

The next problem is that the reports are updated but the message doesn't appear for either user. This is because a Pusher/Push notification is not being sent for this specific IOU event. Notice that any other 'event' (create IOU, send message, cancel IOU, pay IOU) will trigger the following debug message in the other clients console: 

Screenshot 2021-06-02 at 15 54 50

This is how we trigger updates to report comments (and therefore IOU actions), so we'll need to ensure that users are notified as part of the pay-with-expensify flow.

**Comment by jasperhuangg** [comment]
@Julesssss I opened up a Web-E PR that should fix our pusher notification problem. It updates PayWithWallet in the PHP layer to check if the report that was just paid for in Auth is an IOU report, and sends a pusher notification if that's the case. I requested you for review!

https://github.com/Expensify/Web-Expensify/pull/31113

**Review by Julesssss — APPROVED** [review]
Tested well for me. All yours @tylerkaraszewski 

Screenshot 2021-06-03 at 14 25 03

**Review by tylerkaraszewski — APPROVED** [review]

**Comment by roryabraham** [comment]
I think that @kevinksullivan and I were able to successfully QA this on Friday last week. Let me confirm.

**Comment by roryabraham** [comment]
Verified that the front-end is working for me on iOS, Mobile Safari, Chrome, Desktop. Just need someone, I think from @ring1, to confirm that the reimbursements are added to the `reimbursements` table by running this query:

```sql
SELECT * FROM reimbursements WHERE reportID=71125175;
```

**Comment by roryabraham** [comment]
Looks like this _may_ have been a QA-fail, but I'll let @jasperhuangg retest and confirm. I used `Pay With Expensify` for several IOUs on Expensify.cash staging in that report, but [the reimbursements table was empty](https://expensify.slack.com/archives/C03TQ48KC/p1623112168372800).

Nonetheless, since `payWithExpensify` is hidden behind a beta this does not need to be a deploy blocker.

**Pull request merged**